sdk/storage/storage-blob/samples/typescript/proxyAuth.ts

```diff
@@ -0,0 +1,41 @@
+/* 
+ Setup: Enter your storage account name and shared key in main()
+*/
+
+import { BlobServiceClient, StorageClient, SharedKeyCredential } from "../../src"; // Change to "@azure/storage-blob" in your package
+
+async function main() {
+  // Enter your storage account name and shared key
+  const account = "";
+  const accountKey = "";
+
+  // Use SharedKeyCredential with storage account and account key
+  const sharedKeyCredential = new SharedKeyCredential(account, accountKey);
+
+  // Use sharedKeyCredential, tokenCredential or anonymousCredential to create a pipeline
+  const pipeline = StorageClient.newPipeline(sharedKeyCredential, {
+    proxyOptions: { url: "http://localhost:3128" }
```

Comment:
Let's rename this option to `proxy` since it's already in an option type, Options seems redundant. Otherwise LGTM!